opwen_email_client/webapp/actions.py

```diff
@@ -223,3 +229,66 @@ def __call__(self):
         finally:
             if connection is not None:
                 connection.terminate()
+
+
+class ClientRegister(object):
+    def __init__(self, client_name: str, username: str, access_token: str, logger: Logger):
+        self._client_name = client_name
+        self._github_username = username
+        self._github_access_token = access_token
+        self._log = logger or getLogger(__name__)
+
+    @property
+    def client_domain(self):
+        return '{}.{}'.format(self._client_name, 'lokole.ca')
+
+    @property
+    def client_url_create(self):
+        return 'https://{}/api/email/register/'.format('mailserver.lokole.ca')
+
+    @property
+    def client_url_details(self):
+        return 'https://{}/api/email/register/{}'.format('mailserver.lokole.ca', self.client_domain)
+
+    def __call__(self):
+        create_request_payload = dumps({'domain': self.client_domain}).encode('utf-8')
+        create_request = Request(self.client_url_create)
+        create_request.add_header('Content-Type', 'application/json; charset=utf-8')
+        create_request.add_header('Content-Length', str(len(create_request_payload)))
+        create_request.add_header('Authorization', 'Bearer {}'.format(self._github_access_token))
+
+        try:
+            with urlopen(create_request, create_request_payload):  # nosec
+                pass
+        except HTTPError as ex:
+            self._log.exception('Unable to register client {client_name}: [{status_code}] {message}'.format(
+                client_name=self._client_name,
+                status_code=ex.code,
+                message=ex.read().decode('utf-8').strip()
+            ))
+
+        while(True):
```

Comment:
Given that the registration logic can take a while to complete, I would suggest to move it out of the request/response cycle (to avoid issues with the operation being aborted due to request timeout) and run it as a background task via Celery ([see tasks.py](https://github.com/ascoderu/lokole/blob/57bc073c632e0cfd526f544f54b1f0792a205e52/opwen_email_client/webapp/tasks.py)).

Reply:
I was totally thinking celery as well!